core/src/main/scala/kafka/log/OffsetIndex.scala

```diff
@@ -197,3 +197,37 @@ class OffsetIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writabl
   }
 
 }
+
+
+
+/**
+  * A thin wrapper on top of the raw OffsetIndex object to avoid initialization on construction. This defers the OffsetIndex
+  * initialization to the time it gets accessed so the cost of the heavy memory mapped operation gets amortized over time.
+  *
+  * Combining with skipping sanity check for safely flushed segments, the startup time of a broker can be reduced, especially
+  * for the the broker with a lot of log segments
+  *
+  */
+class OffsetIndexGetter(@volatile private var _file: File, baseOffset: Long, maxIndexSize: Int = -1, writable: Boolean = true) {
```

Comment:
Maybe this could be called `LazyOffsetIndex`.

Reply:
Done.